src/mode/modeHandler.ts

```diff
@@ -129,50 +129,63 @@ export class VimSettings {
   useCtrlKeys     = false;
 }
 
+export enum SearchDirection {
+  Forward = 1,
+  Backward = -1
+}
+
 export class SearchState {
+  private static readonly MAX_SEARCH_RANGES = 1000;
+
   /**
    * Every range in the document that matches the search string.
    */
-  public matchRanges: vscode.Range[] = [];
+  private _matchRanges: vscode.Range[] = [];
+  public get matchRanges(): vscode.Range[] {
```

Comment:
The public prop was used elsewhere, but only to read, so I changed it to a public getter. This and some of the other changes are just opinionated cleanup and totally not necessary, so I can revert them if you folks prefer the original way! I just like to restrict public access as much as possible.